src/ore/src/future.rs

```diff
@@ -135,6 +152,62 @@ where
     }
 }
 
+/// The future returned by [`OreFutureExt::spawn_if_canceled`].
+pub struct SpawnIfCanceled<T>
+where
+    T: Send + 'static,
+{
+    inner: Option<Pin<Box<dyn Future<Output = T> + Send>>>,
+}
+
+impl<T> Future for SpawnIfCanceled<T>
+where
+    T: Send + 'static,
+{
+    type Output = T;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<T> {
+        match &mut self.inner {
+            None => panic!("SpawnIfCanceled polled after completion"),
+            Some(f) => match f.as_mut().poll(cx) {
+                Poll::Pending => Poll::Pending,
+                Poll::Ready(res) => {
```

Comment:
It took me a bit to reason through that where we use it, it only hits once the task is completed. Could you adjust the docs/leave a comment to say that this will end the inner future as soon as it turns ready (which for a task means, as soon as it's finished...).

Reply:
Sorry, @antifuchs, I'm afraid I don't know quite what you mean! Once a future returns `Poll::Ready` you're not allowed to call `poll` on it again (unless it happens to be fused), but this doesn't do anything to end the inner future early?

Reply:
Ah, that requirement to not call `poll` again was what I meant - I mixed up fused & un-fused futures in my head, apparently... and phrased the suggestion above weirdly. Somehow in my head, I assumed that the future returning "Ready" would mean it could get polled again. Since that's not the case, I think you can consider this resolved (:

Reply:
Ah, gotcha! Sounds good!